lasagne/nonlinearities.py

```diff
@@ -269,6 +268,76 @@ def elu(x):
     return theano.tensor.switch(x > 0, x, theano.tensor.expm1(x))
 
 
+# selu
+class SELU(object):
+    """Scaled Exponential Linear Unit :math:`\\varphi(x) =
+    \\lambda (x > 0) ? x : \\alpha(e^x-1)`
+
+    The Scaled Exponential Linear Unit (SELU) was introduced in [1]_
+    as an activation function that allows the construction of
+    self-normalizing neural networks.
+
+    Parameters
+    ----------
+    scale : float32
+        The scale parameter :math:`\\lambda` for scaling all output.
+
+    scale_neg  : float32
+        The scale parameter :math:`\\alpha`
+        for scaling output for negative argument values.
+
+    Methods
+    -------
+    __call__(x)
+        Apply the SELU function to the activation `x`.
+
+    Examples
+    --------
+    In contrast to other activation functions in this module, this is
+    a class that needs to be instantiated to obtain a callable:
+
+    >>> from lasagne.layers import InputLayer, DenseLayer
+    >>> l_in = InputLayer((None, 100))
+    >>> from lasagne.nonlinearities import SELU
+    >>> selu = SELU(2, 3)
+    >>> l1 = DenseLayer(l_in, num_units=200, nonlinearity=selu)
+
+    See Also
+    --------
+    selu: Instance with :math:`\\alpha\\approx 1.6733,
+    \\lambda\\approx 1.0507`, as used in [1]_.
+
+    References
+    ----------
+    .. [1] Günter Klambauer, Thomas Unterthiner,
+           Andreas Mayr, Sepp Hochreiter (2017):
+       Self-Normalizing Neural Networks,
+       https://arxiv.org/abs/1706.02515
```

Comment:
I tried to resolve the problem, but I simply can't push the line
``` .. [1] Günter Klambauer, Thomas Unterthiner, Andreas Mayr, Sepp Hochreiter (2017):  ```
shorter than 80 characters, invoking PEP8 check.

Otherwise, I solved all the problems you mentioned. Waiting for your feedback ;)